tests/integration/targets/ec2_vol/tasks/main.yml

```diff
@@ -77,6 +77,7 @@
           - "'instance_id' in volume1.volume.attachment_set"
           - not volume1.volume.attachment_set.instance_id
           - not volume1.volume.encrypted
+          - volume1.volume.tags.ResourcePrefix == "{{ resource_prefix }}"
```

Comment:
It's very easy to mangle the tags (see also https://github.com/ansible-collections/amazon.aws/pull/224#pullrequestreview-569287512) I'd like to see some slightly more comprehensive tests in here.

Reply:
Hi @tremble. Can you please elaborate on what you want?

Reply:
It might make more sense to add the additional tests in #242 as that's where it looks like preserve_tags is being fixed, what do you think @tremble?

Reply:
I'm good with the tag tests being added in either this PR or the next, I'd just like to ensure that they're added especially given we're messing with tagging.  If you try to add them and the results get mangled I'm also good with that just being documented and them actually being added in another PR.

```
      tags:
        "lowercase spaced": 'hello cruel world'
        "Title Case": 'Hello Cruel World'
        CamelCase: 'SimpleCamelCase'
        snake_case: 'simple_snake_case'
```